web/beacon-app/src/features/topics/components/RevokeAPIKeyModal.tsx

```diff
@@ -13,14 +14,47 @@ type RevokeAPIKeyModalProps = {
   onClose: () => void;
 };
 
+// TODO: This component needs to be refactored
```

Comment:
Thanks for adding this. Could you possibly include what should be refactored? It doesn't have to be a long explanation, but I think that a little more detail would be helpful to others reading the code.